src/S3Iterator.cpp

```diff
@@ -1,165 +1,7 @@
 #include "S3Iterator.h"
```

Comment:
Why is this file still here? It doesn't seem to provide any meaningful purpose. I see that S3SparseIterator and S3IteratorLibSVM inherit from S3Iterator, but I think it would be better to just remove S3Iterator altogether since they only inherit a single variable (config).

Reply:
Yes, I think getNext should be pushed up to S3Iterator. Checking if it's possible.